lib/spack/spack/stage.py

```diff
@@ -597,6 +597,11 @@ class DIYStage(object):
     """Simple class that allows any directory to be a spack stage."""
 
     def __init__(self, path):
+        if path is None:
+            raise ValueError("Cannot construct DIYStage without a path.")
+        elif not os.path.isdir(path):
+            raise StagePathError("The stage path directory does not exist.")
```

Comment:
Pass the path as a second parameter so it gets output:

```python
raise StagePathError("The stage path directory does not exist:", path)
```

The SpackError constructor will handle this -- so no need to add code to `StagePathError`.